File: src/test/java/com/alibaba/mtc/perf/tps/CreateThreadLocalInstanceTps.java
package com.alibaba.mtc.perf.tps;

import com.alibaba.mtc.perf.Utils;

/**
 * @author ding.lid
 */
public class CreateThreadLocalInstanceTps {
    public static void main(String[] args) throws Exception {
        TpsCounter tpsCounter = new TpsCounter(2);
        tpsCounter.run(new Runnable() {
            @Override
            public void run() {
                ThreadLocal<String> threadLocal = new ThreadLocal<String>();
                threadLocal.set(Utils.getRandomString());
            }
        });

        while (true) {
            long start = tpsCounter.getCount();
            Thread.sleep(1000);
            System.out.printf("tps: %d\n", tpsCounter.getCount() - start);
        }
    }
}

File: library/src/test/java/com/alibaba/perf/tps/CreateThreadLocalInstanceTps.kt
@file:JvmName("CreateThreadLocalInstanceTps")

package com.alibaba.perf.tps

import com.alibaba.perf.getRandomString

/**
 * @author Jerry Lee (oldratlee at gmail dot com)
 */
fun main() {
    val tpsCounter = TpsCounter(2)

    tpsCounter.setAction(Runnable {
        val threadLocal = ThreadLocal<String>()
        threadLocal.set(getRandomString())
    })

    while (true) {
        val start = tpsCounter.count
        Thread.sleep(1000)
        System.out.printf("tps: %,d\n", tpsCounter.count - start)
    }
}
